### ISSUE TITLE
Bug: WebRTC html stream not loading on Firefox

### ISSUE THREAD

**Issue by conorlap** [AlexxIT/go2rtc#120]
As of latest build v0.1-rc.4 the webrtc html page (http://192.168.0.22:1984/webrtc.html?src=camera.amcrest_ad410_go2rtc) doesn't load the stream on Firefox. It works fine on Chrome.
I've cleared cache and restarted browser.
I rolled back to v0.1-rc.3 and it works fine in Firefox again.

Firefox 107.0 
macOS Ventura 13.0.1

**Comment by NickM-27** [comment]
What do you browser logs say?

**Comment by maxi1134** [comment]
Is your camera h265?

**Comment by AlexxIT** [comment]
You can increase log level and check there

**Comment by conorlap** [comment]
**My config:**

```
streams:
  camera.amcrest_ad410_go2rtc:
    - rtsp://admin:XXXXXXX@192.168.0.25:554/cam/realmonitor?channel=1&subtype=0&#backchannel=0  
  camera.ad410_go2rtc_substream:
    - rtsp://admin:XXXXXXXX@192.168.0.25:554/cam/realmonitor?channel=1&subtype=1&#backchannel=0
  camera.office_go2rtc: 
    - rtsp://192.168.0.8:554/11/user=admin_password=XXXXXXXX
  camera.kitchen_go2rtc:
    - rtsp://admin:XXXXXXXX@192.168.0.14/stream1

  
webrtc:
  listen: ":8555"  # address of your local server (TCP)
  candidates:
    - stun:8555  # if you have dynamic public IP-address

log:
  level: debug
  api: trace
  exec: debug
  ngrok: info
  rtsp: warn
  streams: error
  webrtc: fatal

```






**Go2RTC logs:**
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
09:23:05.643 INF go2rtc version 0.1-rc.4 linux/amd64
09:23:05.643 DBG cwd=/config
09:23:05.643 TRC [api] register path path=/
09:23:05.643 TRC [api] register path path=/api/streams
09:23:05.643 TRC [api] register path path=/api/ws
09:23:05.643 INF [api] listen addr=:1984
09:23:05.644 TRC [api] register path path=/api/devices
09:23:05.644 TRC [api] register path path=/static
09:23:05.644 TRC [api] register path path=/streams
09:23:05.644 TRC [api] register path path=/stream/
09:23:05.644 TRC [api] register path path=/stream
09:23:05.644 TRC [api] register path path=/api/webrtc
09:23:05.644 TRC [api] register path path=/api/frame.mp4
09:23:05.644 TRC [api] register path path=/api/stream.mp4
09:23:05.644 TRC [api] register path path=/api/frame.jpeg
09:23:05.644 TRC [api] register path path=/api/stream.mjpeg
09:23:05.644 INF [srtp] listen addr=:8443
09:23:05.645 TRC [api] register path path=/api/homekit
09:23:05.645 TRC [api] register path path=/api/stack
09:23:05.645 TRC [api] register path path=/api/exit
09:23:10.986 TRC [api] GET /webrtc.html?src=camera.amcrest_ad410_go2rtc
09:23:11.034 TRC [api] GET /api/ws?src=camera.amcrest_ad410_go2rtc
09:23:22.782 TRC [api] GET /
09:23:24.797 TRC [api] GET /webrtc.html?src=camera.ad410_go2rtc_substream
09:23:24.831 TRC [api] GET /api/ws?src=camera.ad410_go2rtc_substream
```


**Browser logs:**
```
Uncaught (in promise) DOMException: Cannot set ICE candidate for level=<Nothing> mid=: No such transceiver. 15
pc.onicecandidate 
RTCIceCandidate { candidate: "candidate:1 2 UDP 1686052862 84.65.171.149 63157 typ srflx raddr 0.0.0.0 rport 0", sdpMid: "0", sdpMLineIndex: 0, usernameFragment: "d92741aa" }
[webrtc.html:61:21](http://192.168.0.22:1984/webrtc.html?src=camera.amcrest_ad410_go2rtc)
pc.onicecandidate 
RTCIceCandidate { candidate: "", sdpMid: "0", sdpMLineIndex: 0, usernameFragment: "d92741aa" }
[webrtc.html:61:21](http://192.168.0.22:1984/webrtc.html?src=camera.amcrest_ad410_go2rtc)
pc.onicecandidate null [webrtc.html:61:21](http://192.168.0.22:1984/webrtc.html?src=camera.amcrest_ad410_go2rtc)
ws.onmessage 
Object { type: "webrtc/candidate", value: "candidate:4179490071 1 udp 1694498815 84.65.171.149 51562 typ srflx raddr 0.0.0.0 rport 51562" }
[webrtc.html:48:21](http://192.168.0.22:1984/webrtc.html?src=camera.amcrest_ad410_go2rtc)
Uncaught (in promise) DOMException: Cannot set ICE candidate for level=<Nothing> mid=: No such transceiver.
WebRTC: ICE failed, add a TURN server and see about:webrtc for more details
```

**Comment by calisro** [comment]
It doesn't appear to work in firefox for me either.  stream.html works but webrtc.html does not.

**Comment by calisro** [comment]
idk if this helps but in about:webrtc in firefox, we can see this issue:

```
PeerConnection ID:{9d8a37e0-18b6-4f43-bd93-2e926555e147} 1670338564409344 (id=6442450949 url=http://xxxx:1984/webrtc.html?src=front
RTCConfiguration
bundlePolicy: balanced
iceTransportPolicy: all
peerIdentity(Provided/Not Provided): Not Provided
sdpSemantics: Not Provided
iceServers:
urls: ["stun:stun.l.google.com:19302"]
credential(Provided/Not Provided): Not Provided
userName(Provided/Not Provided): Not Provided
```

**Comment by markfrancisonly** [comment]
can confirm this is still a problem in firefox 107 as of v0.1-rc.5. 

the webrtc.html js client can establish a connection only when loaded from the same host as go2rtc.

when the webrtc js client is loaded from a host external to go2rtc, a valid remote candidate address is never established and firefox reports "WebRTC: ICE failed, add a TURN server and see about:webrtc for more details"

**Comment by AlexxIT** [comment]
Fixed in v0.1-rc.6

**Comment by forresthopkinsa** [comment]
Currently observing this behavior in v1.9.9

> WebRTC: ICE failed, add a TURN server and see about:webrtc for more details
